handle-mulipart-form-data/backend_server/server.bal

```diff
@@ -0,0 +1,67 @@
+import ballerina/http;
+import ballerina/io;
+import ballerina/log;
+import ballerina/mime;
+import ballerina/random;
+
+type CustomerRegistrationData record {|
+    string firstName;
+    string lastName;
+    string address;
+    int dependents;
+|};
+
+final table<CustomerRegistrationData> customerTable = table [
+    {firstName: "James", lastName: "Clerk", address: "456, ElmAvenue, Suite-7, Willowville", dependents: 2},
+    {firstName: "John", lastName: "Doe", address: "789, OakLane, Unit-C12, Pineville", dependents: 6},
+    {firstName: "Anna", lastName: "Watson", address: "101, Maple-Road, OfficeD, Birchwood", dependents: 7}
+];
+
+@http:ServiceConfig {
+    cors: {
+        allowOrigins: ["*"]
+    }
+}
+service /crm on new http:Listener(9090) {
+    resource function get customers() returns CustomerRegistrationData[] {
+        return customerTable.toArray();
+    }
+
+    resource function get customers/[string customerId]/agreement() returns byte[]|http:NotFound {
+        byte[]|error agreementForm = getAgreementForm(customerId);
+        if agreementForm is error {
+            return <http:NotFound>{body: {message: "Agreement form not found for the customer ID: " + customerId}};
+        }
+        return agreementForm;
+    }
+
+    resource function post customers(http:Request request) returns http:Created|http:InternalServerError {
+        do {
+            mime:Entity[] bodyParts = check request.getBodyParts();
+            string registrationDataString = check bodyParts[0].getText();
+            json registrationDataJson = check registrationDataString.fromJsonString();
+            CustomerRegistrationData registrationData = check registrationDataJson.cloneWithType();
+            byte[] agreemntForm = check bodyParts[1].getByteArray();
+            byte[] image = check bodyParts[2].getByteArray();
+            string customerId = check registerCustomer(registrationData, agreemntForm, image);
+            return <http:Created>{body: {message: "Customer registered successfully.", customerId}};
```

Comment:
I think it's better to store customerId in the table and also store agreementForm and image in the file system against the customerId (e.g. <customerId>_agreement.pdf, <customerId>_image.png). Then we can return the actual agreement in the /customers/<id>/agreement resource.

Reply:
Updated.